planning/behavior_path_planner/src/scene_module/pull_out/pull_out_module.cpp

```diff
@@ -424,97 +425,78 @@ void PullOutModule::planWithPriorityOnEfficientPath(
+    if (i == start_pose_candidates.size() - 1) return false;
 
-  // check if start pose candidates are valid
-  if (start_pose_candidates.empty()) {
-    return;
-  }
+    // check next path if back is needed
+    const auto & pull_out_start_pose_next = start_pose_candidates.at(i + 1);
+    const auto pull_out_path_next = planner->plan(pull_out_start_pose_next, goal_pose);
+    // not found safe path
+    if (!pull_out_path_next) {
+      return false;
+    }
 
-  for (size_t i = 0; i < start_pose_candidates.size(); i++) {
-    status_.back_finished = i == 0;
-    const auto & pull_out_start_pose = start_pose_candidates.at(i);
-    // plan with each planner
+    // Update status variables with the next path information
+    status_.is_safe = true;
+    status_.pull_out_path = *pull_out_path_next;
+    status_.pull_out_start_pose = pull_out_start_pose_next;
+    status_.planner_type = planner->getPlannerType();
+    return true;
+  };
+
+  using PriorityOrder = std::vector<std::pair<size_t, std::shared_ptr<PullOutPlannerBase>>>;
+  const auto makePriorityOrder_PlannerFirst = [&]() {
+    PriorityOrder order_priority;
     for (const auto & planner : pull_out_planners_) {
-      planner->setPlannerData(planner_data_);
-      const auto pull_out_path = planner->plan(pull_out_start_pose, goal_pose);
-      // not found safe path
-      if (!pull_out_path) {
-        continue;
+      for (size_t i = 0; i < start_pose_candidates.size(); i++) {
+        order_priority.emplace_back(i, planner);
       }
-      // use current path if back is not needed
-      if (status_.back_finished) {
-        status_.is_safe = true;
-        status_.pull_out_path = *pull_out_path;
-        status_.pull_out_start_pose = pull_out_start_pose;
-        status_.planner_type = planner->getPlannerType();
-        break;
-      }
-
-      if (i == start_pose_candidates.size() - 1) continue;
+    }
+    return order_priority;
+  };
 
-      //  check next path if back is needed
-      const auto & pull_out_start_pose_next = start_pose_candidates.at(i + 1);
-      const auto pull_out_path_next = planner->plan(pull_out_start_pose_next, goal_pose);
-      // not found safe path
-      if (!pull_out_path_next) {
-        continue;
+  const auto makePriorityOrder_PoseFirst = [&]() {
+    PriorityOrder order_priority;
+    for (size_t i = 0; i < start_pose_candidates.size(); i++) {
+      for (const auto & planner : pull_out_planners_) {
+        order_priority.emplace_back(i, planner);
       }
-      status_.is_safe = true;
-      status_.pull_out_path = *pull_out_path_next;
-      status_.pull_out_start_pose = pull_out_start_pose_next;
-      status_.planner_type = planner->getPlannerType();
-      break;
-    }
-    if (status_.is_safe) {
-      break;
     }
+    return order_priority;
+  };
+
+  // Choose loop order based on priority_on_efficient_path
+  PriorityOrder order_priority;
+  if (priority_on_efficient_path) {
+    order_priority = makePriorityOrder_PlannerFirst();
```

Comment:
I'm not sure but is this function name is ok in terms of naming convention?

Reply:
changed to `make_loop_order_planner/pose_first`